assets/mdk/proof-dependencies.mdk

```diff
@@ -14,28 +14,36 @@ Css Header:
 
-<!-- inline-dafny unspecified/BinarySearch -->
+<!-- inline-dafny redundant/BinarySearch -->
 
-As a final example, proof dependency analysis can also detect preconditions, assumptions, or intermediate assertions that are unnecessary for constructing a proof of correctness. These redundant specifications may indicate a mistake, such that they would not be redundant if written correctly. Or it may be possible to remove them, simplifying the program and proof process. In the following example, the precondition (which one might think would be required to make the code well-defined) turns out to be unnecessary.
+As a final example, if the specification doesn't cover all of the program's behavior, any implementation will satisfy the uncovered portions. This latter case is frequently intentional -- you may only want to prove certain simple properties about your program, such as the lack of runtime errors -- but it can be useful to confirm that the unspecified portions do not include code you intended to specify.
 
-<!-- inline-dafny redundant/BinarySearch -->
+In the following example of a binary search implementation, we removed the redundant assumption, and Dafny verifies only the absence of undefined behavior. In particular, it doesn't prove that the return value correctly corresponds to what you'd expect from a binary search implementation. Therefore, any of the return statements could be modified without causing a verification failure.
+
+<!-- inline-dafny unspecified/BinarySearch -->
 
 # Proof dependencies in Dafny
 
+With v4.4.0[^1] Dafny will include features to automatically detect problems like contradictory specifications, redundant assumptions, and unspecified code, using a notion of _proof dependency_. A proof dependency is a portion of a program that needed to be taken into account in order to construct a proof that the program is correct. When talking about proof dependencies, let's use the term _property_ more formally from here on to refer a particular statement about a Dafny program that we want to prove. This might arise from an `assert` statement, an `ensures` clause, or a `requires` clause of a call, for example. Let's use _assumption_ to refer to anything that's assumed, arising, for example, from an `assume` statement, a `requires` clause, or an `ensures` clause of a call.
+
+[^1]: Or any nightly build after 2023-10-24.
+
+Proof dependencies make it possible to identify properties proved due to contradictory assumptions. To legitimately prove that a property holds of the program, it must be the case that the proof takes the property itself into account. If it's possible to construct a proof that the property holds _without_ considering what it is we're ultimately trying to prove, we say that we've proved it _vacuously_. This can occur because the assumptions in scope at that program location are contradictory, or that that location is unreachable.
```

Comment:
```suggestion
Proof dependencies make it possible to identify properties proved due to contradictory assumptions. To legitimately prove that a property holds of the program, it must be the case that the proof takes the property itself into account. If it's possible to construct a proof that the property holds _without_ considering what we're ultimately trying to prove, we say that we've proved it _vacuously_. This can occur because the assumptions in scope at that program location are contradictory, or that that location is unreachable.
```